src/gui/tagfilterwidget.cpp

```diff
@@ -0,0 +1,225 @@
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * In addition, as a special exception, the copyright holders give permission to
+ * link this program with the OpenSSL project's "OpenSSL" library (or with
+ * modified versions of it that use the same license as the "OpenSSL" library),
+ * and distribute the linked executables. You must obey the GNU General Public
+ * License in all respects for all of the code used other than "OpenSSL".  If you
+ * modify file(s), you may extend this exception to your version of the file(s),
+ * but you are not obligated to do so. If you do not wish to do so, delete this
+ * exception statement from your version.
+ */
+
+#include "tagfilterwidget.h"
+
+#include <QAction>
+#include <QDebug>
+#include <QHeaderView>
+#include <QLayout>
+#include <QMenu>
+#include <QMessageBox>
+
+#include "base/bittorrent/session.h"
+#include "base/utils/misc.h"
+#include "autoexpandabledialog.h"
+#include "guiiconprovider.h"
+#include "tagfiltermodel.h"
+#include "tagfilterproxymodel.h"
+
+namespace
+{
+    QString getTagFilter(const TagFilterProxyModel *const model, const QModelIndex &index)
+    {
+        QString tagFilter; // Defaults to All
+        if (index.isValid()) {
+            if (!index.parent().isValid() && (index.row() == 1))
+                tagFilter = "";  // Untagged
+            else if (index.parent().isValid() || (index.row() > 1))
+                tagFilter = model->tag(index);
+        }
+
+        return tagFilter;
+    }
+}
+
+TagFilterWidget::TagFilterWidget(QWidget *parent)
+    : QTreeView(parent)
+{
+    TagFilterProxyModel *proxyModel = new TagFilterProxyModel(this);
+    proxyModel->setSortCaseSensitivity(Qt::CaseInsensitive);
+    proxyModel->setSourceModel(new TagFilterModel(this));
+    setModel(proxyModel);
+    setFrameShape(QFrame::NoFrame);
+    setHorizontalScrollBarPolicy(Qt::ScrollBarAlwaysOff);
+    setVerticalScrollBarPolicy(Qt::ScrollBarAlwaysOff);
+    setSizePolicy(QSizePolicy::Expanding, QSizePolicy::Fixed);
+    setUniformRowHeights(true);
+    setHeaderHidden(true);
+    setIconSize(Utils::Misc::smallIconSize());
+#if defined(Q_OS_MAC)
+    setAttribute(Qt::WA_MacShowFocusRect, false);
+#endif
+    setContextMenuPolicy(Qt::CustomContextMenu);
+    sortByColumn(0, Qt::AscendingOrder);
+    setCurrentIndex(model()->index(0, 0));
+
+    connect(this, &TagFilterWidget::collapsed, this, &TagFilterWidget::callUpdateGeometry);
+    connect(this, &TagFilterWidget::expanded, this, &TagFilterWidget::callUpdateGeometry);
+    connect(this, &TagFilterWidget::customContextMenuRequested, this, &TagFilterWidget::showMenu);
+    connect(selectionModel(), &QItemSelectionModel::currentRowChanged, this
+            , &TagFilterWidget::onCurrentRowChanged);
```

Comment:
This is a subjective opinion (I don't remember that we had something about this in the coding style, though there is something similar in the initialization list of constructors). Personally last time I prefer to start all line breaks from the operator or comma.